source/agora/script/Engine.d

```diff
@@ -735,6 +755,144 @@ public class Engine
+
+        Checks floating-transaction signatures for use with the Flash layer.
+
+        Verifies the sequence signature by blanking the input, reading the
+        minimum sequence, the key, the new sequence, and the signature off
+        the stack and validates the signature.
+
+        If any of the arguments expected on the stack are missing,
+        an error string is returned.
+
+        The `sig_valid` parameter will be set to the validation result
+        of the signature.
+
+        Params:
+            OP = the opcode
+            stack = should contain the Signature and Public Key
+            tx = the transaction that should have been signed
+            input = the associated Input to blank when signing
+            sig_valid = will contain the signature validation result
+
+        Returns:
+            an error string if the needed arguments on the stack are missing,
+            otherwise returns null
+
+    ***************************************************************************/
+
+    private string verifySequenceSignature (OP op)(ref Stack stack,
+        in Transaction tx, in Input input, out bool sig_valid)
+        nothrow @safe //@nogc  // stack.pop() is not @nogc
+    {
+        static assert(op == OP.CHECK_SEQ_SIG || op == OP.VERIFY_SEQ_SIG);
+
+        // if changed, check assumptions
+        static assert(Point.sizeof == 32);
+        static assert(Signature.sizeof == 64);
+
+        // top to bottom: <min_seq> <key> <new_seq> <sig>
+        // lock script typically pushes <min_seq> <key>
+        // while the unlock script pushes <new_seq> <sig>
+        if (stack.count() < 4)
+        {
+            static immutable err1 = op.to!string
+                ~ " opcode requires 4 items on the stack";
+            return err1;
+        }
+
+        const min_seq_bytes = stack.pop();
+        if (min_seq_bytes.length != ulong.sizeof)
+        {
+            static immutable err2 = op.to!string
+                ~ " opcode requires 8-byte minimum sequence on the stack";
+            return err2;
+        }
+
+        const ulong min_sequence = littleEndianToNative!ulong(
+            min_seq_bytes[0 .. ulong.sizeof]);
+
+        const key_bytes = stack.pop();
+        if (key_bytes.length != Point.sizeof)
+        {
+            static immutable err3 = op.to!string
+                ~ " opcode requires 32-byte public key on the stack";
+            return err3;
+        }
+
+        const Point point = Point(key_bytes);
+        if (!point.isValid())
+        {
+            static immutable err4 = op.to!string
+                ~ " 32-byte public key on the stack is invalid";
+            return err4;
+        }
+
+        const seq_bytes = stack.pop();
+        if (seq_bytes.length != ulong.sizeof)
+        {
+            static immutable err5 = op.to!string
+                ~ " opcode requires 8-byte sequence on the stack";
+            return err5;
+        }
```

Comment:
I think the test code for seq_bytes is missing.
However, the length of `min_seq_bytes` has been tested.

Reply:
Good eye! I'll add the test.